jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py

```diff
@@ -91,49 +111,62 @@ def __init__(
         self._prefetch_data = []
 
         self.logger = get_logger(self)
+        self._load_dump(dump_path, dump_func, prefetch_size, **kwargs)
+
+    def _load_dump(self, dump_path, dump_func, prefetch_size, **kwargs):
```

Comment:
This function became to complex and should be split into sub-functions. Same is true for `_train`. Actually, I think we should redesign this a bit. Let me have some thinking about it.